src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonOutputFormatter.cs

```diff
@@ -65,13 +66,17 @@ public sealed override async Task WriteResponseBodyAsync(OutputFormatterWriteCon
                 // the behavior you get when the user does not declare the return type and with Json.Net at least at the top level.
                 var objectType = context.Object?.GetType() ?? context.ObjectType;
                 await JsonSerializer.SerializeAsync(writeStream, context.Object, objectType, SerializerOptions);
+                if (writeStream is TranscodingWriteStream transcodingStream)
```

Comment:
A little hacky. Can you add giant comment with the reasoning?

Reply:
Done

Reply:
While I have you here, @GrabYourPitchforks wanted to find out if Kestrel's streams override the `Memory<byte>` overloads on Stream. His suggestion was to use the `Array` backed APIs if that wasn't the case since it's a bit quicker.

Reply:
Yes they do. Use them 😬